salt/client/__init__.py

```diff
@@ -882,6 +882,7 @@ def cmd_iter(
             else:
                 if kwargs.get(u'yield_pub_data'):
                     yield pub_data
+                kwargs.setdefault('expect_minions', True)
```

Comment:
I would rather see something like this implemented as a new kwarg defined in the function's argspec. That way, the default behavior here can be set to True, but when other functions call `cmd_iter`, we can account for any potential behavior changes.

Specifically, I am worried about a behavior change in `salt/daemons/masterapi.py` and `salt/cli/batch.py` files and where `expect_minions` is already defaulted to `False` in `get_iter_returns`.